samples/snippets/src/main/java/com/example/bigquery/QueryExternalTableAws.java

```
@@ -0,0 +1,89 @@
+    String tableName = "MY_TABLE_NAME";
+    // Create a aws connection
+    // projects/{project_id}/locations/{location_id}/connections/{connection_id}
+    String connectionId = "MY_CONNECTION_NAME";
+    String sourceUri = "s3://your-bucket-name/";
+    CsvOptions options = CsvOptions.newBuilder().setSkipLeadingRows(1).build();
+    Schema schema =
+        Schema.of(
+            Field.of("name", StandardSQLTypeName.STRING),
+            Field.of("post_abbr", StandardSQLTypeName.STRING));
+    String query =
+        String.format(
+            "SELECT * FROM s%:%s.%s WHERE name LIKE 'W%%'", projectId, datasetName, tableName);
+    ExternalTableDefinition externalTable =
+        ExternalTableDefinition.newBuilder(sourceUri, options)
+            .setConnectionId(connectionId)
+            .setSchema(schema)
+            .build();
+    queryExternalTableAws(projectId, datasetName, tableName, externalTable, query);
+  }
+
+  public static void queryExternalTableAws(
+      String projectId,
+      String datasetName,
+      String tableName,
+      ExternalTableDefinition externalTable,
+      String query) {
+    try {
+      // Initialize client that will be used to send requests. This client only needs to be created
+      // once, and can be reused for multiple requests.
+      BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+
+      TableId tableId = TableId.of(projectId, datasetName, tableName);
+      // Create a permanent table linked to the Aws file
+      bigquery.create(TableInfo.of(tableId, externalTable));
+
+      // Example query to find states starting with 'W'
+      TableResult results = bigquery.query(QueryJobConfiguration.of(query));
+
+      results
+          .iterateAll()
+          .forEach(row -> row.forEach(val -> System.out.printf("%s,", val.toString())));
+
+      System.out.println("Query on aws external permanent table performed successfully.");
+    } catch (BigQueryException | InterruptedException e) {
```

Great that your showing BigQueryException | InterruptedException here, but should we just propagate them, rather than catching and printing stuff?